pytesmo/validation_framework/metric_calculators.py

```diff
@@ -236,3 +240,725 @@ def calc_metrics(self, data, gpi_info):
+        mse_dict = mse._asdict()
+        mse_corr_dict = mse_corr._asdict()
+        mse_bias_dict = mse_bias._asdict()
+        mse_var_dict = mse_var._asdict()
+
+        # calulcate tau
+        if self.calc_tau:
+            tau, p_tau = df_metrics.kendalltau(data)
+            tau_dict = tau._asdict()
+            p_tau_dict = p_tau._asdict()
+        else:
+            tau = p_tau = p_tau_dict = tau_dict = None
+
+        # todo: should the data be scaled before calculating the ubRMSD?
+        #data_scaled = scale(data, method='mean_std')
+        # calculate ubRMSD
+        ubRMSD_nT = df_metrics.ubrmsd(data)
+        ubRMSD_dict = ubRMSD_nT._asdict()
+
+        # get single dataset metrics
+        # calculate SNR
+        x = data[self.df_columns[0]].values[subset]
+        y = data[self.df_columns[1]].values[subset]
+        z = data[self.df_columns[2]].values[subset]
+
+        snr, err, beta = metrics.tcol_snr(x, y, z)
+
+        for i, name in enumerate(self.ds_names):
+            dataset['{:}_snr'.format(name)][0] = snr[i]
+            dataset['{:}_err_var'.format(name)][0] = err[i]
+            dataset['{:}_beta'.format(name)][0] = beta[i]
+
+
+        for tds_name in self.tds_names:
+            R = pearson_R[tds_name]
+            p_R = pearson_p[tds_name]
+            rho = spea_rho[tds_name]
+            p_rho = spea_p[tds_name]
+            bias = bias_dict[tds_name]
+            mse = mse_dict[tds_name]
+            mse_corr = mse_corr_dict[tds_name]
+            mse_bias = mse_bias_dict[tds_name]
+            mse_var = mse_var_dict[tds_name]
+            rmsd = rmsd_dict[tds_name]
+            ubRMSD = ubRMSD_dict[tds_name]
+            if tau_dict and p_tau_dict:
+                tau = tau_dict[tds_name]
+                p_tau = p_tau_dict[tds_name]
+
+
+            split_tds_name = tds_name.split('_and_')
+            tds_name_key = "{:}_{:}".format(self.ds_names_lut[
+                split_tds_name[0]],
+                self.ds_names_lut[
+                split_tds_name[1]])
+
+            dataset['{:}_R'.format(tds_name_key)][0] = R
+            dataset['{:}_p_R'.format(tds_name_key)][0] = p_R
+            dataset['{:}_rho'.format(tds_name_key)][0] = rho
+            dataset['{:}_p_rho'.format(tds_name_key)][0] = p_rho
+            dataset['{:}_bias'.format(tds_name_key)][0] = bias
+            dataset['{:}_mse'.format(tds_name_key)][0] = mse
+            dataset['{:}_mse_corr'.format(tds_name_key)][0] = mse_corr
+            dataset['{:}_mse_bias'.format(tds_name_key)][0] = mse_bias
+            dataset['{:}_mse_var'.format(tds_name_key)][0] = mse_var
+            dataset['{:}_rmsd'.format(tds_name_key)][0] = rmsd
+            dataset['{:}_ubRMSD'.format(tds_name_key)][0] = ubRMSD
+
+            if self.calc_tau:
+                dataset['{:}_tau'.format(tds_name_key)][0] = tau
+                dataset['{:}_p_tau'.format(tds_name_key)][0] = p_tau
+
+        return dataset
+
+
+
+if __name__  == '__main__':
```

Comment:
this can be removed

Reply:
ok, thanks, @sebhahn have you noticed that here the wrong R is used? https://github.com/wpreimes/pytesmo/blob/d715209417c6165f599d7e9848b7ca10b52a2b19/pytesmo/validation_framework/metric_calculators.py#L482

Reply:
No, could you fix this? Is not a big problem though, because we do not use spearman

Reply:
sure, already fixed it